src/catalog/trigger_catalog.cpp

```diff
@@ -0,0 +1,318 @@
+  static std::unique_ptr<TriggerCatalog> trigger_catalog(
+      new TriggerCatalog(txn));
+
+  return trigger_catalog.get();
+}
+
+TriggerCatalog::TriggerCatalog(concurrency::Transaction *txn)
+    : AbstractCatalog("CREATE TABLE " CATALOG_DATABASE_NAME
+                      "." TRIGGER_CATALOG_NAME
+                      " ("
+                      "trigger_oid           INT NOT NULL, "
+                      "trigger_name          VARCHAR NOT NULL, "
+                      "database_oid          INT NOT NULL, "
+                      "table_oid             INT NOT NULL, "
+                      "trigger_type          INT NOT NULL, "
+                      "fire_condition        VARCHAR, "
+                      "function_name         VARCHAR, "
+                      "function_arguments    VARCHAR, "
+                      "time_stamp            INT NOT NULL);",
+                      txn) {
+  // Add secondary index here if necessary
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"database_oid", "table_oid", "trigger_type"}, TRIGGER_CATALOG_NAME "_skey0",
+      false, IndexType::BWTREE, txn);
+
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"database_oid", "table_oid"}, TRIGGER_CATALOG_NAME "_skey1",
+      false, IndexType::BWTREE, txn);
+
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"trigger_name", "database_oid", "table_oid"}, TRIGGER_CATALOG_NAME "_skey2",
+      false, IndexType::BWTREE, txn);
+
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"trigger_oid"}, TRIGGER_CATALOG_NAME "_skey3",
+      false, IndexType::BWTREE, txn);  
+}
+
+TriggerCatalog::~TriggerCatalog() {}
+
+bool TriggerCatalog::InsertTrigger(
+    std::string trigger_name, oid_t database_oid, oid_t table_oid,
+    int16_t trigger_type,
+    std::string fire_condition, //TODO: this actually should be expression
+    std::string function_name,
+    std::string function_arguments,
+    int64_t time_stamp, type::AbstractPool *pool,
+    concurrency::Transaction *txn) {
+
+
+  std::unique_ptr<storage::Tuple> tuple(
+      new storage::Tuple(catalog_table_->GetSchema(), true));
+
+
+  LOG_INFO("type of trigger inserted:%d", trigger_type);
+
+  auto val0 = type::ValueFactory::GetIntegerValue(oid_++ | TRIGGER_OID_MASK);
+  auto val1 = type::ValueFactory::GetVarcharValue(trigger_name, pool);
+  auto val2 = type::ValueFactory::GetIntegerValue(database_oid);
+  auto val3 = type::ValueFactory::GetIntegerValue(table_oid);
+  auto val4 = type::ValueFactory::GetIntegerValue(trigger_type);
+  auto val5 = type::ValueFactory::GetVarcharValue(fire_condition, pool);
+  auto val6 = type::ValueFactory::GetVarcharValue(function_name, pool);
+  auto val7 = type::ValueFactory::GetVarcharValue(function_arguments, pool);
+  auto val8 = type::ValueFactory::GetIntegerValue(time_stamp);
+
```

Comment:
Thanks for pointing it out. We are following the naming style of other catalogs. "val0" means the value for the first column.